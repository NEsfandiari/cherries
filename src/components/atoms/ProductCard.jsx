import React, { Component } from 'react'
import styled from 'styled-components'

const Container = styled.div`
  text-align: center;
  margin-bottom: 70px;
  width: 187.02px;
  flex: 1 0 25%;

  .product-card-image {
    width: 187.02px;
    height: 238.63px;
    margin-bottom: 0;
  }
  .product-card-name {
    color: #000000;
    font-family: Glacial Indifference;
    font-size: 16px;
    font-weight: 700;
    line-height: 20px
    width: 168px;
    text-align: left;
    margin-bottom: 0;
  }
  .product-card-price {
    color: #444444;
    font-family: Glacial Indifference;
    font-size: 16px;
    font-weight: 400;
    line-height: 20px;
    width: 58px;
    text-align: left;
  }
`

class ProductCard extends Component {
  render() {
    const product = this.props.product

    return (
      <Container>
        <div className="product-card">
          <img
            className="product-card-image"
            src={product.images[0].file.url}
            alt="earring"
          />
          <p className="product-card-name">{product.title}</p>
          <p className="product-card-price">${product.price}</p>
        </div>
      </Container>
    )
  }
}

export default ProductCard
