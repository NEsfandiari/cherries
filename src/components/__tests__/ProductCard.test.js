import React from 'react'
import renderer from 'react-test-renderer'
import ProductCard from '../atoms/ProductCard'

describe('ProductCard', () =>
  it('renders correctly', () => {
    const product = {
      title: 'Astronaut Baby',
      price: 22.99,
      images: [
        {
          file: {
            url:
              '//images.ctfassets.net/4yejk6rho2rp/60XuYBiY6IIes0aqyAOQIs/e177ec00483171ad7d8a71e6cf8f462f/fashion-woman-dangle-earrings-heartly-space-dream-tong-qu-the-earth-rocket-astronauts-blue-pendants-earrings.jpg_640x640.jpg',
          },
        },
      ],
    }
    const tree = renderer.create(<ProductCard product={product} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
