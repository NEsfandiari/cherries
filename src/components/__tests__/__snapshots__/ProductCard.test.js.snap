// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `
<div
  className="sc-bdVaJa eNKvBx"
>
  <a
    className="product-card"
    href="Astronaut-Baby/"
  >
    <img
      alt="earring"
      className="product-card-image"
      src="//images.ctfassets.net/4yejk6rho2rp/60XuYBiY6IIes0aqyAOQIs/e177ec00483171ad7d8a71e6cf8f462f/fashion-woman-dangle-earrings-heartly-space-dream-tong-qu-the-earth-rocket-astronauts-blue-pendants-earrings.jpg_640x640.jpg"
    />
    <p
      className="product-card-name"
    >
      Astronaut Baby
    </p>
    <p
      className="product-card-price"
    >
      $
      22.99
    </p>
  </a>
</div>
`;
